src/canvas/Canvas.ts

```diff
@@ -1495,158 +1509,99 @@ export class Canvas extends SelectableCanvas {
-    // this._hoveredTargets = [];
-    // this._hoveredTargets = this.targets.concat();
-    this._setActiveObject(newActiveSelection, e);
-    this._fireSelectionEvents([currentActive], e);
-  }
+      this._activeSelection.removeAll();
+      resetObjectTransform(this._activeSelection);
+      this._activeSelection.add(
+        ...(groupingTarget.isInFrontOf(activeObject)
+          ? [activeObject, groupingTarget]
+          : [groupingTarget, activeObject])
+      );
 
-  /**
-   * Finds objects inside the selection rectangle and group them
-   * @private
-   * @param {Event} e mouse event
-   */
-  _groupSelectedObjects(e: TPointerEvent) {
-    const group = this._collectObjects(e);
-    // do not create group for 1 element only
-    if (group.length === 1) {
-      this.setActiveObject(group[0], e);
-    } else if (group.length > 1) {
-      const aGroup = new ActiveSelection(group.reverse(), {
-        canvas: this,
-      });
-      this.setActiveObject(aGroup, e);
+      this._hoveredTarget = this._activeSelection;
+      // ISSUE 4115: should we consider subTargets here?
+      // this._hoveredTargets = [];
+      // this._hoveredTargets = this.targets.concat();
+      this._setActiveObject(this._activeSelection, e);
+      this._fireSelectionEvents([activeObject], e);
     }
   }
 
-  /**
-   * @private
-   */
-  _collectObjects(e: TPointerEvent) {
-    const group: FabricObject[] = [],
-      _groupSelector = this._groupSelector,
+  protected _collectObjects(e: TPointerEvent) {
+    const _groupSelector = this._groupSelector,
       point1 = new Point(_groupSelector.ex, _groupSelector.ey),
       point2 = point1.add(new Point(_groupSelector.left, _groupSelector.top)),
-      selectionX1Y1 = point1.min(point2),
-      selectionX2Y2 = point1.max(point2),
-      allowIntersect = !this.selectionFullyContained,
+      tl = point1.min(point2),
+      br = point1.max(point2),
+      size = br.subtract(tl),
       isClick = point1.eq(point2);
-    // we iterate reverse order to collect top first in case of click.
-    for (let i = this._objects.length; i--; ) {
-      const currentObject = this._objects[i];
-
-      if (
-        !currentObject ||
-        !currentObject.selectable ||
-        !currentObject.visible
-      ) {
-        continue;
-      }
-
-      if (
-        (allowIntersect &&
-          currentObject.intersectsWithRect(
-            selectionX1Y1,
-            selectionX2Y2,
-            true
-          )) ||
-        currentObject.isContainedWithinRect(
-          selectionX1Y1,
-          selectionX2Y2,
-          true
-        ) ||
-        (allowIntersect &&
-          currentObject.containsPoint(selectionX1Y1, undefined, true)) ||
-        (allowIntersect &&
-          currentObject.containsPoint(selectionX2Y2, undefined, true))
-      ) {
-        group.push(currentObject);
-        // only add one object if it's a click
-        if (isClick) {
-          break;
-        }
-      }
-    }
```

Comment:
comments like: `moved this in a collection mixin called `collectObject`` Help me save entire minutes.

Reply:
I didn't write the description yet
sorry